libminifi/src/core/ProcessGroup.cpp

```diff
@@ -91,33 +91,16 @@ ProcessGroup::~ProcessGroup() {
   }
 }
 
-bool ProcessGroup::isRootProcessGroup() {
-  return (type_ == ROOT_PROCESS_GROUP);
-}
-
 bool ProcessGroup::isRemoteProcessGroup() {
   return (type_ == REMOTE_PROCESS_GROUP);
 }
 
 
-void ProcessGroup::addProcessor(const std::shared_ptr<Processor>& processor) {
+void ProcessGroup::addProcessor(std::unique_ptr<Processor>&& processor) {
```

Comment:
Consider pass-by-value and a precondition to check that it's not null

Reply:
What kind of precondition check were you thinking of? A simple `if` or some `gsl`?

Reply:
Done.